agent/api/task/task.go

```diff
@@ -2195,10 +2195,20 @@ func (task *Task) generateENIExtraHosts() []string {
 
 	extraHosts := []string{}
 
-	for _, ip := range eni.GetIPV4Addresses() {
-		host := fmt.Sprintf("%s:%s", hostname, ip)
-		extraHosts = append(extraHosts, host)
+	if eni.IPv6Only() {
+		// Add IPv6 addresses only if task ENI is IPv6-only.
+		// Historically, IPv6 addresses are not added for dual stack ENIs.
+		for _, ip := range eni.GetIPV6Addresses() {
+			host := fmt.Sprintf("%s:%s", hostname, ip)
+			extraHosts = append(extraHosts, host)
+		}
+	} else {
+		for _, ip := range eni.GetIPV4Addresses() {
+			host := fmt.Sprintf("%s:%s", hostname, ip)
+			extraHosts = append(extraHosts, host)
+		}
```

Comment:
We could possibly simplify this check by setting a range parameter and then using that to drive what extraHosts are populated:

```
// default case we pull from eni's IPv4 addresses
ipAddresses = eni.GetIPV4Addresses()
// Use IPv6 addresses only if task ENI is IPv6-only.
// Historically, IPv6 addresses are not added for dual stack ENIs.
if eni.IPv6Only() {
    ipAddresses = eni.GetIPV6Addresses()
}

for _, ip := range eni.GetIPV6Addresses() {
    host := fmt.Sprintf("%s:%s", hostname, ip)
    extraHosts = append(extraHosts, host)
}
```